src/components/Reports/PartnerGivingAnalysisReport/PartnerGivingAnalysisReport.tsx

```diff
@@ -161,27 +163,16 @@ export const PartnerGivingAnalysisReport: React.FC<Props> = ({
         >
           <CircularProgress data-testid="LoadingPartnerGivingAnalysisReport" />
         </Box>
-      ) : contacts.length ? (
-        <>
-          {!staffAccountLoading && staffAccountData?.staffAccount?.id ? (
-            <BalanceCard
-              donationPeriodTotalSum={
-                data?.partnerGivingAnalysis?.donationPeriodTotalSum
-              }
-            />
-          ) : null}
-          <Table
-            data={contacts}
-            totalCount={data?.partnerGivingAnalysis.totalCount ?? 0}
-            onSelectOne={toggleSelectionById}
-            isRowChecked={isRowChecked}
-            paginationModel={paginationModel}
-            handlePageChange={handlePageChange}
-            sortModel={sortModel}
-            handleSortChange={handleSortChange}
-          />
-        </>
-      ) : (
+      )}
+      {contacts.length > 0 && !loading && (
```

Comment:
There is still flickering while loading. I think you might want this so that we render the table as long as we have some contacts to render. It's OK if we're still loading the rest of the contacts.

```suggestion
      {contacts.length > 0 && (
```

Reply:
I didn't notice how bad the flickering is. I tried removing this line but still have the same issue. It seems like the contacts updates from `useFetchAllPages` causes the table to re-render each call. Do you know a work around for this?